File: app/src/main/java/guepardoapps/mycoins/utils/StringHelper.java
package guepardoapps.mycoins.utils;

import android.support.annotation.NonNull;

public class StringHelper {
    public static int GetStringCount(
            @NonNull String stringToTest,
            @NonNull String charToFind) {
        int lastIndex = 0;
        int count = 0;

        while (lastIndex != -1) {
            lastIndex = stringToTest.indexOf(charToFind, lastIndex);
            if (lastIndex != -1) {
                count++;
                lastIndex += charToFind.length();
            }
        }

        return count;
    }

    public static int[] GetCharPositions(
            @NonNull String stringToTest,
            char charToFind) {
        int count = GetStringCount(stringToTest, String.valueOf(charToFind));
        if (count <= 0) {
            return new int[]{};
        }

        int[] positions = new int[count];
        int foundIndex = 0;

        for (int charIndex = 0; charIndex < stringToTest.length(); charIndex++) {
            if (stringToTest.charAt(charIndex) == charToFind) {
                positions[foundIndex] = charIndex;
                foundIndex++;
            }
        }

        return positions;
    }

    public static boolean StringsAreEqual(@NonNull String[] stringArray) {
        boolean areEqual = true;

        int stringCount = stringArray.length;

        for (int index = 1; index < stringCount; index++) {
            areEqual &= stringArray[index] == stringArray[index - 1];
        }

        return areEqual;
    }

    public static String SelectString(
            @NonNull String[] stringArray,
            @NonNull String stringToFound) {
        int stringCount = stringArray.length;

        for (int index = 1; index < stringCount; index++) {
            if (stringArray[index].contains(stringToFound)) {
                if (!stringArray[index].contains("Error")) {
                    return stringArray[index];
                }
            }
        }

        return "";
    }
}

File: app/src/main/java/guepardoapps/mycoins/utils/StringHelper.kt
package guepardoapps.mycoins.utils

import android.support.annotation.NonNull

class StringHelper {
    companion object {
        fun getStringCount(@NonNull test: String, @NonNull find: String): Int {
            var lastIndex = 0
            var count = 0

            while (lastIndex != -1) {
                lastIndex = test.indexOf(find, lastIndex)
                if (lastIndex != -1) {
                    count++
                    lastIndex += find.length
                }
            }

            return count
        }

        fun getCharPositions(@NonNull test: String, char: Char): List<Int> {
            return test
                    .mapIndexed { index, c -> if (c == char) index else -1 }
                    .filter { x -> x != -1 }
        }

        fun stringsAreEqual(@NonNull array: Array<String>): Boolean {
            return array.all { x -> array.all { y -> x == y } }
        }

        fun excludeAndSelectString(@NonNull array: Array<String>, @NonNull exclude: String, @NonNull find: String): String? {
            return array.find { x -> !x.contains(exclude) && x.contains(find) }
        }
    }
}